common/src/main/java/io/netty/util/ThreadDeathWatcher.java

```diff
@@ -77,31 +78,22 @@
      * @throws IllegalArgumentException if the specified {@code thread} is not alive
      */
     public static void watch(Thread thread, Runnable task) {
-        if (thread == null) {
-            throw new NullPointerException("thread");
-        }
-        if (task == null) {
-            throw new NullPointerException("task");
-        }
+        ObjectUtil.checkNotNull(thread, "thread");
+
         if (!thread.isAlive()) {
             throw new IllegalArgumentException("thread must be alive.");
         }
 
-        schedule(thread, task, true);
+        schedule(thread, ObjectUtil.checkNotNull(task, "task"), true);
```

Comment:
move the mull check under the null check for thread to ensure we not change behaviour.